huggingface/transformers
# Adding AutoTokenizer and AutoModel classes that automatically detect architecture - Clean up tokenizers

As discussed in #890 

Classes that automatically detect the relevant model/config/tokenizer to instantiate based on the`pretrained_model_name_or_path` string provided to `AutoXXX.from_pretrained(pretrained_model_name_or_path)`.

Right now:
- `AutoConfig`
- `AutoTokenizer`
- `AutoModel` (bare models outputting hidden-states)

Missing:
- Tests
- Maybe a few other architectures beside raw models (`AutoModelWithLMHead`, `AutoModelForSequenceClassification`, `AutoModelForTokensClassification`, `AutoModelForQuestionAnswering`)
- Check if we can make hubconfs simpler to maintain using AutoModels.

Additional stuff:
- add a `unk_token` to GPT2 to fix #799
- clean up tokenizers and associated tests
